### PR TITLE
feat(readme): explaining the oracle creation from the factory

### PR DESCRIPTION
**Pull request by tomrpl**
Updating the readme, thanks to the `MorphoChainlinkOracleV2Factory`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
-
 Morpho Blue Oracles are contracts that can be used as oracles for markets on [Morpho Blue](https://github.com/morpho-org/morpho-blue).
 The oracles implement the `IOracle` interface defined in [`IOracle.sol`](https://github.com/morpho-org/morpho-blue/blob/main/src/interfaces/IOracle.sol#L9): they return the price of 1 asset of collateral token quoted in 1 asset of loan token.
 
-## ChainlinkOracle
+## MorphoChainlinkOracleV2
 
-The `ChainlinkOracle` is an oracle that uses Chainlink-compliant feeds to provide price data.
+The `MorphoChainlinkOracleV2` is an oracle that uses Chainlink-compliant feeds to provide price data.
```
Comment (MerlinEgalite) [inline README.md:8]:
```suggestion
The `MorphoChainlinkOracleV2` is an oracle that uses Chainlink-interfacce-compliant feeds to provide price data.
```

File: README.md [hunk 2]
```diff
@@ -18,6 +18,34 @@ This oracle handles the following cases among others (let's say that our pair is
 - A/C and C/B are feeds (typically, WBTC/BTC and BTC/ETH).
 - A'/C and B/C are feeds, and there is an exchange rate between A and A'. (typically A=sDAI and A'=DAI).
 
+## WstETH/stETH Exchange Rate Adapter
+
+A specific implementation, the WstEthStEthExchangeRateChainlinkAdapter, provides the exchange rate between wstETH and stETH using Chainlink. This adapter is designed for precise, decentralized price feeds on Ethereum, ensuring compatibility with the MorphoChainlinkOracleV2 for markets involving these assets.
```
Comment (MerlinEgalite) [inline README.md:23]:
```suggestion
A specific implementation, the `WstEthStEthExchangeRateChainlinkAdapter`, provides the exchange rate between wstETH and stETH.
```

File: README.md [hunk 3]
```diff
@@ -18,6 +18,34 @@ This oracle handles the following cases among others (let's say that our pair is
...
+
+Deployed on Ethereum Mainnet, this adapter can be found in Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code). It's recommended for use as a reliable price feed source in Morpho oracles.
```
Comment (MerlinEgalite) [inline README.md:25]:
```suggestion
Deployed on Ethereum Mainnet, this adapter can be found on Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
```

File: README.md [hunk 4]
```diff
@@ -18,6 +18,34 @@ This oracle handles the following cases among others (let's say that our pair is
...
+
+## MorphoChainlinkOracleV2Factory
+
+The `MorphoChainlinkOracleV2Factory` is a factory that deploys `MorphoChainlinkOracleV2` instances.
+
+The factory facilitates the easy creation and indexing of MorphoChainlinkOracleV2 oracles. Developed by Morpho Labs, this factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need for deploying individual contracts manually for each pair.
```
Comment (MerlinEgalite) [inline README.md:31]:
```suggestion
The factory facilitates the easy creation and indexing of `MorphoChainlinkOracleV2` oracles. This factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need for deploying individual contracts manually for each pair.
```

File: README.md [hunk 5]
```diff
@@ -18,6 +18,34 @@ This oracle handles the following cases among others (let's say that our pair is
...
+
+### Usage
+
+The Ethereum Mainnet address of this factory is [0x3A7bB36Ee3f3eE32A60e9f2b33c1e5f2E83ad766](https://etherscan.io/address/0x3a7bb36ee3f3ee32a60e9f2b33c1e5f2e83ad766#code).
+
+To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the createMorphoChainlinkOracleV2 function with the following parameters:
```
Comment (MerlinEgalite) [inline README.md:37]:
```suggestion
To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the `createMorphoChainlinkOracleV2` function with the following parameters:
```

File: README.md [hunk 6]
```diff
@@ -28,22 +56,54 @@ Run test: `forge test`
 
 ## Deploy an Oracle
```
Comment (MerlinEgalite) [inline README.md:57]: can be moved to usage IMO

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -18,32 +18,92 @@ This oracle handles the following cases among others (let's say that our pair is
 - A/C and C/B are feeds (typically, WBTC/BTC and BTC/ETH).
 - A'/C and B/C are feeds, and there is an exchange rate between A and A'. (typically A=sDAI and A'=DAI).
 
-## Getting Started
+## WstETH/stETH Exchange Rate Adapter
 
-Install dependencies: `forge install`
+A specific implementation, the `WstEthStEthExchangeRateChainlinkAdapter`, provides the exchange rate between wstETH and stETH.
```
Comment (Rubilmax) [inline README.md:23]:
```suggestion
A specific implementation, the `WstEthStEthExchangeRateChainlinkAdapter`, provides the exchange rate between wstETH and stETH as a Chainlink-interface-compliant feed.
```

File: README.md [hunk 8]
```diff
@@ -18,32 +18,92 @@ This oracle handles the following cases among others (let's say that our pair is
...
 
-In a `.env` file, set `ETH_RPC_URL` to the URL of an Ethereum provider.
+Deployed on Ethereum Mainnet, this adapter can be found on Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
```
Comment (Rubilmax) [inline README.md:25]:
```suggestion
This adapter is deployed on Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
```

File: README.md [hunk 9]
```diff
@@ -18,32 +18,92 @@ This oracle handles the following cases among others (let's say that our pair is
...
 
-Run test: `forge test`
+## MorphoChainlinkOracleV2Factory
+
+The `MorphoChainlinkOracleV2Factory` is a factory that deploys `MorphoChainlinkOracleV2` instances.
+
+The factory facilitates the easy creation and indexing of `MorphoChainlinkOracleV2` oracles. This factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need for deploying individual contracts manually for each pair.
```
Comment (Rubilmax) [inline README.md:31]:
```suggestion
The factory facilitates the creation and indexing of `MorphoChainlinkOracleV2` oracles. This factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need to deploy individual contracts manually for each pair.
```

File: README.md [hunk 10]
```diff
@@ -18,32 +18,92 @@ This oracle handles the following cases among others (let's say that our pair is
...
+- `baseVaultConversionSample`:1,
+- `baseFeed1`: 0x0000000000000000000000000000000000000000,
+- `baseFeed2`: 0x0000000000000000000000000000000000000000,
+- `baseTokenDecimals`:18,
+- `quoteVault`:0x0000000000000000000000000000000000000000,
+- `quoteVaultConversionSample`:1,
+- `quoteFeed1`: 0xEe9F2375b4bdF6387aa8265dD4FB8F16512A1d46,
+- `quoteFeed2`: 0x0000000000000000000000000000000000000000,
+- `quoteTokenDecimals`:6,
+- `salt`: <user-defined value used to make the address unique>,
```
Comment (Rubilmax) [inline README.md:65]:
```suggestion
```json
{
  "baseVault": "0x0000000000000000000000000000000000000000",
  "baseVaultConversionSample": 1,
  "baseFeed1": "0x0000000000000000000000000000000000000000",
  "baseFeed2": "0x0000000000000000000000000000000000000000",
  "baseTokenDecimals": 18,
  "quoteVault": "0x0000000000000000000000000000000000000000",
  "quoteVaultConversionSample": 1,
  "quoteFeed1": "0xEe9F2375b4bdF6387aa8265dD4FB8F16512A1d46",
  "quoteFeed2": "0x0000000000000000000000000000000000000000",
  "quoteTokenDecimals": 6,
  "salt": "user-defined 32 bytes used to make the address unique",
```

```

**Review by MerlinEgalite — DISMISSED** [review]

File: README.md [hunk 11]
```diff
@@ -18,32 +18,98 @@ This oracle handles the following cases among others (let's say that our pair is
...
+## WstETH/stETH Exchange Rate Adapter
 
-Install dependencies: `forge install`
+A specific implementation, the `WstEthStEthExchangeRateChainlinkAdapter`, provides the exchange rate between wstETH and stETH as a Chainlink-interface-compliant feed.
 
-In a `.env` file, set `ETH_RPC_URL` to the URL of an Ethereum provider.
+This adapter is deployed on Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
 
-Run test: `forge test`
+## MorphoChainlinkOracleV2Factory
```
Comment (MathisGD) [inline README.md:27]: I would not do a separated category for the factory, and just add this to the corresponding oracle part

File: README.md [hunk 12]
```diff
@@ -18,32 +18,98 @@ This oracle handles the following cases among others (let's say that our pair is
...
 - A'/C and B/C are feeds, and there is an exchange rate between A and A'. (typically A=sDAI and A'=DAI).
 
-## Getting Started
+## WstETH/stETH Exchange Rate Adapter
 
-Install dependencies: `forge install`
+A specific implementation, the `WstEthStEthExchangeRateChainlinkAdapter`, provides the exchange rate between wstETH and stETH as a Chainlink-interface-compliant feed.
 
-In a `.env` file, set `ETH_RPC_URL` to the URL of an Ethereum provider.
+This adapter is deployed on Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
```
Comment (MathisGD) [inline README.md:25]: there is no need to say "mainnet" all the time imo
```suggestion
This adapter is deployed on Ethereum at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
```

File: README.md [hunk 13]
```diff
@@ -18,32 +18,98 @@ This oracle handles the following cases among others (let's say that our pair is
 - A/C and C/B are feeds (typically, WBTC/BTC and BTC/ETH).
 - A'/C and B/C are feeds, and there is an exchange rate between A and A'. (typically A=sDAI and A'=DAI).
```
Comment (MathisGD) [inline README.md:19]: this has to be adapted

File: README.md [hunk 14]
```diff
@@ -18,32 +18,98 @@ This oracle handles the following cases among others (let's say that our pair is
...
+- `quoteFeed1`: "0xEe9F2375b4bdF6387aa8265dD4FB8F16512A1d46",
+- `quoteFeed2`: "0x0000000000000000000000000000000000000000",
+- `quoteTokenDecimals`: 6,
+- `salt`: <user-defined value used to make the address unique>,
 ```
 
+and for the sDAI/USDC oracle:
+
+```json
+- `baseVault`: "0x83f20f44975d03b1b09e64809b757c47f942beea",
```
Comment (QGarchery) [inline README.md:72]:
```suggestion
- `baseVault`: "0x83F20F44975D03b1b09e64809B757c47f942BEeA",
```
to have the correct checksum, as it is the case for the other addresses

**Review by QGarchery — DISMISSED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**